cli/internal/images/push.go

```diff
@@ -24,19 +24,18 @@ func PushAll(imageTarballPath string, buildImageList []string) {
 			return
 		}
 
-		onlineName, err := docker.ParseNormalizedNamed(src)
-		if err != nil {
-			logContext.Debug(err)
-			logContext.Warn("Unable to parse the image domain")
-			return
-		}
+		offlineName := SwapHost(src, targetHost)
 
-		offlineName := strings.Replace(src, docker.Domain(onlineName), config.ZarfLocalIP, 1)
-		logrus.Info(offlineName)
 		err = crane.Push(img, offlineName, cranePlatformOptions)
 		if err != nil {
 			logContext.Debug(err)
 			logContext.Warn("Unable to push the image to the registry")
 		}
 	}
 }
+
+func SwapHost(src string, targetHost string) string {
+	var parser = regexp.MustCompile(`(?im)^([a-z0-9\-.]+\.[a-z0-9\-]+:?[0-9]*)?/?(.+)$`)
```

Comment:
Can we add a human readable docstring that describes what this regex is doing?

Reply:
that's very human-readable sir!!!  j/k sorry regex is one of things, it reads pretty simple once you know it will, I'll do you one better I link to an example you can test with that explains each element.

Reply:
Please check it now and tell me what you think.

Reply:
That's awesome!